Server Crash: Choose new entity action Bugfix

Cheaterpaul opened the pull request:
now skips actions with weight 0
now writes EntityClassType to nbt
#396

Cheaterpaul committed:
bugfix

Cheaterpaul committed:
static EntityClassType

maxanier commented on src/api/java/de/teamlapen/vampirism/api/entity/EntityClassType.java:
```diff
@@ -35,4 +40,19 @@ public AttributeModifier getDamageModifier() {
+    }
+
+    public static EntityClassType getEntityClassType(int id) {
+        return ID.get(id);
```
To be sure this should probably be checked against map.size()  and return a default if to large

Cheaterpaul replied:
dont set an classtype if id is corrupt(entityclasstype is set by contructor)

maxanier commented on src/main/java/de/teamlapen/vampirism/entity/action/EntityActionHandler.java:
```diff
@@ -139,13 +142,20 @@ private void updateHandler() {
+                entry.add(new EntityActionEntry(weight, e));
+            }
+        }
+        if (weightsum > 0) {
```
`weightsum` is never increased :)

Cheaterpaul committed:
safety additions

now it dosen't dont change the entityclasstype if nbt id is corrupt
increases weightsum
